ksql-engine/src/main/java/io/confluent/ksql/planner/plan/JoinNode.java

```diff
@@ -177,15 +180,20 @@ public SchemaKStream buildStream(final StreamsBuilder builder,
 
   }
 
-  private SchemaKTable tableForJoin(
+  // package private for test
+  SchemaKTable tableForJoin(
       final StreamsBuilder builder,
       final KsqlConfig ksqlConfig,
       final KafkaTopicClient kafkaTopicClient,
       final MetastoreUtil metastoreUtil,
       final FunctionRegistry functionRegistry,
       final Map<String, Object> props) {
 
-    final SchemaKStream schemaKStream = right.buildStream(builder, ksqlConfig, kafkaTopicClient, metastoreUtil, functionRegistry, props);
+    Map<String, Object> joinTableProps = new HashMap<>();
+    joinTableProps.putAll(props);
+    joinTableProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "EARLIEST");
```

Comment:
probably be better to expose this as a KSQLConfig property - and default it to EARLIEST. There might be some edge cases where tables are massive and cannot be bootstrapped etc

Reply:
`auto.offset.reset` is configurable for KSQL, however, this one is a special case for Stream-Table join and is required to always be `earliest` and if we have it as `latest` the join would be incorrect.

Reply:
Thats what I mean - this is a special case and we dont always know what people will do with the system. It is possible that there will be an instance where the hardcoded 'earliest' is not wanted. exposing a new property 'table.auto.offset.reset' would work around this.

Reply:
I see what you mean but in this case `earliest` is the only correct option and setting it to anything else will result in incorrect query results. So even if we have `table.auto.offset.reset` it's value cannot be changed and should always be `earliest` even when we have the table change topic containing huge number of rows.